homeassistant/components/ridwell/__init__.py

```diff
@@ -0,0 +1,84 @@
+"""The Ridwell integration."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+
+from aioridwell import async_get_client
+from aioridwell.client import RidwellAccount, RidwellPickupEvent
+from aioridwell.errors import InvalidCredentialsError, RidwellError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DATA_ACCOUNT, DATA_COORDINATOR, DOMAIN, LOGGER
+
+DEFAULT_UPDATE_INTERVAL = timedelta(hours=1)
+
+PLATFORMS: list[str] = ["sensor"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Ridwell from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {}
```

Comment:
It's better to not store things in hass.data until after we know we won't raise ConfigEntryNotReady etc.